### DISCUSSION TITLE
How to handle (smallish) contributions that do not fit into an existing repository?

### DISCUSSION THREAD

**Discussion by benediktahrens** [UniMath/UniMath#1719] [General]
Some students are working on porting (smallish) formalizations from other libraries to UniMath. Several of these formalizations use HITs, and hence don't fit into UniMath/UniMath. They don't seem to fit into any of the existing satellites, either.

Any suggestions for how to store these contributions? How about a repository "Contrib", where each such project becomes a package? (Possibly some of the formalizations can be moved later, into a more topical repository.)

I would be very grateful for any suggestions.

**Comment by nmvdw** [comment]
I am in favor of having a "Contrib" satellite. I see the following advantages:
- It allows for broader usage of UniMath.
- User contributions will be maintained.
- Smaller (student) projects are useful and very welcome.

The main challenge comes from an organizational perspective: how do we guarantee that "Contrib" remains usable? For example, if there are many small developments that are in conflict with each other, then it might become difficult to use/maintain. This problem would especially annoying when later students want to build forth upon already existing work.

This problem is solvable, but this does mean that sometimes, the maintainers need to do some work in maintaining the code (reorganization/refactoring), and that sometimes we might need to be a bit strict with PRs.

**Comment by rmatthes** [comment]
Use of HIT is one argument why such material would have to be in a satellite. But how would you think about not respecting the UniMath coding style for convenience, e.g., by using the inductive families and type classes even when, in principle, code in UniMath coding style would be possible? The remark by @nmvdw that the problems grow when later projects want to build on the "Contrib" satellite would even be more serious for such parts.

**Comment by nmvdw** [comment]
My personal opinion for these two would be:
- You can freely use inductive families. I think this one is quite important, because they are a feature missing in UniMath, which is very useful in many applications.
- Be hesitant with using type classes. Debugging them can be challenging and, in my experiences, often there are good alternative methods as well.

I would say that there should be a clear coding style, and that each part would need to be discussed.
